### PR TITLE
GH-104: add missing next/app

### LINKED ISSUE #104: Add missing next/app

### PR DESCRIPTION
**Pull request by halitogunc**
Closes #104

### CODE REVIEW

File: npm/next/7.0/@n4jsd/next/lib/app.n4jsd [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import  * as React from "react";
+import {NextPageContext} from "lib/dist/server/next";
+import {INextRouter} from "lib/router";
+
+export external public interface ~AppInitialProps {
+    pageProps: any;
```
Comment (dbo) [inline npm/next/7.0/@n4jsd/next/lib/app.n4jsd]: - public
- any or better at least Object?

File: npm/next/7.0/@n4jsd/next/lib/app.n4jsd [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+export external public class ~NextAppContext extends NextPageContext {
+    public Component: (props: ~Object with {router: Object}) => React.Element<?>;
```
Comment (dbo) [inline npm/next/7.0/@n4jsd/next/lib/app.n4jsd]: don'T we have a type for Router?

File: npm/next/7.0/@n4jsd/next/lib/app.n4jsd [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    public ctx: NextPageContext;
+    public AppTree: any;
+}
+
+export default external public class NextApp <PropsT extends React.ComponentProps, StateT extends Object>
+  extends React.Component<PropsT, StateT> {
+
+    public static getInitialProps(ctx: NextAppContext): Promise<AppInitialProps, any>;
+    @Override
+    public componentDidCatch(error: any, info: ~Object with { componentStack }): void;
```
Comment (dbo) [inline npm/next/7.0/@n4jsd/next/lib/app.n4jsd]: why the override?

File: npm/next/7.0/@n4jsd/next/lib/dist/server/next.n4jsd [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import {IncomingMessage} from "http";
 import {ServerResponse} from "http";
 import {Socket} from "net";
-import {INextRouter} from "next/router";
+import {INextRouter} from "lib/router";
```
Comment (dbo) [inline npm/next/7.0/@n4jsd/next/lib/dist/server/next.n4jsd]: lib?

File: npm/next/7.0/@n4jsd/next/lib/prefetch.n4jsd [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import * as React from "react";
-import { LinkProps } from "link";
+import { LinkProps } from "lib/link";
```
Comment (dbo) [inline npm/next/7.0/@n4jsd/next/lib/prefetch.n4jsd]: lib? or next/lib/link? it misses the npm

**Review by dbo — APPROVED** [review]
with comments resolved

File: npm/next/7.0/@n4jsd/next/lib/prefetch.n4jsd (hunk 5 above)
Comment (halitogunc) [reply npm/next/7.0/@n4jsd/next/lib/prefetch.n4jsd]: This is a local file, not a library.

File: npm/next/7.0/@n4jsd/next/lib/dist/server/next.n4jsd (hunk 4 above)
Comment (halitogunc) [reply npm/next/7.0/@n4jsd/next/lib/dist/server/next.n4jsd]: IDE resolves INextRouter incorrectly as "router" for `app.n4jsd`. This is why we have an outer lib folder.

**Pull request merged**